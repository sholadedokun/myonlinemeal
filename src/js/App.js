import React, { Component } from 'react';
import Header from './components/header';
import {BrowserRouter as Router,  Route } from 'react-router-dom';
import {Grid, Row} from 'react-bootstrap'
import Footer from './components/footer';
import Home from './components/home'
import Dashboard from './components/dashboard'
import Howitworks from './components/howitworks'
import Ourmission from './components/ourmission'
import Whymom from './components/whymom'
import Help from './components/help'
import requireAuth from './components/auth/require_auth';
import { connect } from 'react-redux';
import { getAllPlans } from './actions/planActions'
export class App extends Component {
    componentWillMount(){
        this.props.getAllPlans().then(data=>{
        })
    }
    render() {
        return (
            <Router>
                <Grid fluid={true} className="App">
                    <Header className="App-header"></Header>
                    <Route  exact path="/"  component={Home} />
                    <Route  exact path="/dashboard"   component={requireAuth(Dashboard)}  />
                    <Route  exact path="/Howitworks"  component={Howitworks} />
                    <Route  exact path="/ourmission"  component={Ourmission} />
                    <Route  exact path="/whymom"  component={Whymom} />
                    <Route  exact path="/help"  component={Help} />
                    <Footer />
                </Grid>
            </Router>
        );
    }
}

export default connect(null, {getAllPlans})(App);
