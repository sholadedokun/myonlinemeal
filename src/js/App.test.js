
import React, { Component } from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import App from './App'
// import Header from './components/header'
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'
// import {addInputs,subtractInputs} from '../src/js/actions/calculatorActions'
import {createStore} from 'redux'
// import calculatorReducers from '../src/js/reducers/calculatorReducers'

const initialState = {user:{error:'', authenticated:'', isOpen:false, page:'login'}}
   const mockStore = configureStore()
   let store,wrapper

   beforeEach(()=>{
       store = mockStore(initialState)
       // wrapper = mount( <Provider store={store}><App /></App> )
   })



// Snapshot for APP React Component

describe('>>>APP --- Snapshot',()=>{
    it('+++capturing Snapshot of APP', () => {
        const renderedValue =  renderer.create(<Provider store={store}><App /></Provider>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});

describe('>>>APP --- Shallow Render REACT COMPONENTS',()=>{
    let wrapper
    beforeEach(()=>{
        wrapper = mount(<Provider store={store}><App /></Provider>)
    })

    it('+++ render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });
    it('+++ render the Header AND FOOTER component', () => {
       expect(wrapper.find('Header').length).toEqual(1)
       expect(wrapper.find('Footer').length).toEqual(1)
    });
    it('+++ Loads the Modal When Login is clicked', () => {
        let header=wrapper.find('Header')
        console.log(header.find('NavItem#login').debug())
        wrapper.find('NavItem#login').simulate('click');
        console.log(wrapper.debug())
       expect(wrapper.find('.loginModal').length).toEqual(1)
       expect(wrapper.find('Footer').length).toEqual(1)
    });
})
