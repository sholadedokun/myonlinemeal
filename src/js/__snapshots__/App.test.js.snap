// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>APP --- Snapshot +++capturing Snapshot of APP 1`] = `
<div
  className="App container-fluid"
>
  <div
    className="header row"
  >
    <nav
      className="navbar navbar-inverse"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <a
            className="navbar-brand"
            href="#"
          >
            MOM
          </a>
          <button
            className="navbar-toggle collapsed"
            onClick={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <ul
            className="nav navbar-nav"
            role={null}
          >
            <li
              className="active"
              role="presentation"
              style={undefined}
            >
              <a
                href="/"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="/howitworks"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                How it Works
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="/help"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                id="login"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                login
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                id="register"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                register
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="row"
  >
    <div
      className="banner "
    >
      <div
        className="register col-md-4 col-xs-12"
      >
        <div
          className="headings"
        >
          <h2
            style={
              Object {
                "fontSize": "1.8em",
                "marginBottom": undefined,
              }
            }
          >
            Chef Cooked, Healthy Meal Delivered to you.
          </h2>
          
          <p />
        </div>
        <input
          name="email"
          placeholder="Your Email Address"
          type="text"
        />
        <button>
          GET STARTED
        </button>
        <p>
          Already have an account? 
          <a
            onClick={[Function]}
          >
            Login Here
          </a>
          <br />
          By continuing, you are agreeing to our 
          <a>
            terms and condition
          </a>
        </p>
      </div>
    </div>
    <div
      className="section col-xs-12"
    >
      <div
        className="headings"
      >
        <h2
          style={
            Object {
              "fontSize": "1.8em",
              "marginBottom": "2em",
            }
          }
        >
          How It Works
        </h2>
        
        <p />
      </div>
      <div
        className="row"
      >
        <div
          className="category col-md-4 col-xs-12"
        >
          <span
            className="fa fa-spoon"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <div
            className="headings"
          >
            <h3
              style={
                Object {
                  "fontSize": "1.2em",
                  "marginBottom": undefined,
                }
              }
            >
              You Choose A Meal Plan
            </h3>
            
            <p />
          </div>
          Simple Text to briefly explain what this category is meant to do, this paragraph should not be more than 25 words.
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="category col-md-4 col-xs-12"
        >
          <span
            className="fa fa-fire"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <div
            className="headings"
          >
            <h3
              style={
                Object {
                  "fontSize": "1.2em",
                  "marginBottom": undefined,
                }
              }
            >
              Our Chef Cooks the Meal
            </h3>
            
            <p />
          </div>
          Simple Text to briefly explain what this category is meant to do, this paragraph should not be more than 25 words.
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="category col-md-4 col-xs-12"
        >
          <span
            className="fa fa-truck"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <div
            className="headings"
          >
            <h3
              style={
                Object {
                  "fontSize": "1.2em",
                  "marginBottom": undefined,
                }
              }
            >
              Meal Delivered, You Enjoy 
            </h3>
            
            <p />
          </div>
          Simple Text to briefly explain what this category is meant to do, this paragraph should not be more than 25 words.
        </div>
      </div>
    </div>
    <div
      className="section mealPlan row"
    >
      <div
        className="details col-xs-12"
      >
        <div
          className="headings"
        >
          <h2
            style={
              Object {
                "fontSize": "1.8em",
                "marginBottom": "2em",
              }
            }
          >
            Our Awesome Meal Plans
          </h2>
          
          <p />
        </div>
        <p>
          Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue eu vulpu magna erat. Aliquam erat volutpat. Nam dui mi tincidunt quis accumsan porttitor facilisis luctus, metus lorem feugiat tempus adipiscing.
        </p>
        <ul>
          <li
            className="active"
          >
            3 Meals
          </li>
          <li>
            5 Meals
          </li>
          <li>
            20 Meals
          </li>
        </ul>
      </div>
      <div
        className=""
      >
        <div
          className="tabbedContent row"
        >
          Nam dui mi tincidunt quis  accumsan porttitor facilisis luctus, metus lorem feugiat tempus adipiscing. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue eu vulpu magna erat. Aliquam erat volutpat.
        </div>
      </div>
    </div>
    <div
      className="section ourchef col-xs-12"
    >
      <div
        className="headings"
      >
        <h2
          style={
            Object {
              "fontSize": "1.8em",
              "marginBottom": "2em",
            }
          }
        >
          Our Chefs Cooks every meal
        </h2>
        
        <p />
      </div>
    </div>
    <div
      className="section col-xs-12"
    >
      <div
        className="headings"
      >
        <h2
          style={
            Object {
              "fontSize": "1.8em",
              "marginBottom": "2em",
            }
          }
        >
          What People are saying
        </h2>
        
        <p />
      </div>
      <div
        className="comment col-md-3 col-sm-6 col-xs-12"
      >
        <div>
          <span
            className="fa fa-star"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <span
            className="fa fa-star"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <span
            className="fa fa-star"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <span
            className="fa fa-star"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
          <span
            className="fa fa-star"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          />
           
        </div>
        <div
          className="headings"
        >
          <h3
            style={
              Object {
                "fontSize": "1.2em",
                "marginBottom": undefined,
              }
            }
          >
            Comment that changed Lives
          </h3>
          
          <p />
        </div>
        <span>
          Some lovely comment about how the food has changed their lives, how it help save money, how it's an healthy choice and how good the meals tastes.
        </span>
        <span
          className="commenter"
        >
          olushola Adedokun
        </span>
      </div>
    </div>
  </div>
  <div
    className="section footer row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="headings"
      >
        <h3
          style={
            Object {
              "fontSize": "1.2em",
              "marginBottom": "1em",
            }
          }
        >
          Fresh, Healthy, Affordable, Convinient Meal
        </h3>
        
        <p />
      </div>
      <p>
        Cool Stuffs to say about our business model and why customers should subscribe to our services and follow our us on social media.
      </p>
      <span
        className="fa fa-facebook"
        style={
          Object {
            "fontSize": "1.2em",
          }
        }
      />
      <span
        className="fa fa-twitter"
        style={
          Object {
            "fontSize": "1.2em",
          }
        }
      />
      <span
        className="fa fa-instagram"
        style={
          Object {
            "fontSize": "1.2em",
          }
        }
      />
    </div>
    <div
      className="hrule col-xs-12"
    />
    <div
      className="col-xs-12"
    >
      <div
        className="headings"
      >
        <h3
          style={
            Object {
              "fontSize": "1.2em",
              "marginBottom": "1em",
            }
          }
        >
          About
        </h3>
        
        <p />
      </div>
      <ul>
        <li>
          <a>
            How it Works
          </a>
        </li>
        <li>
          <a>
            Our Mission
          </a>
        </li>
        <li>
          <a>
            Why MOM
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hrule col-xs-12"
    />
    <div
      className="col-xs-12"
    >
      <div
        className="headings"
      >
        <h3
          style={
            Object {
              "fontSize": "1.2em",
              "marginBottom": "1em",
            }
          }
        >
          Account
        </h3>
        
        <p />
      </div>
      <ul>
        <li>
          <a>
            Sign Up
          </a>
        </li>
        <li>
          <a>
            Your Account
          </a>
        </li>
        <li>
          <a>
            Login
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hrule col-xs-12"
    />
    <div
      className="col-xs-12"
    >
      <div
        className="headings"
      >
        <h3
          style={
            Object {
              "fontSize": "1.2em",
              "marginBottom": "1em",
            }
          }
        >
          Contact Us
        </h3>
        
        <p />
      </div>
      <p>
        Write compelling statement to make people contact us immediately using the channels we are have provided below.
      </p>
      <div
        className="headings"
      >
        <h1
          style={
            Object {
              "fontSize": "2.5em",
              "marginBottom": "1.5em",
            }
          }
        >
          0818-801-9666
        </h1>
        
        <p />
      </div>
    </div>
    <div
      className="copyright col-xs-12"
    >
      © MyOnlineMeals Limited.
    </div>
  </div>
</div>
`;
